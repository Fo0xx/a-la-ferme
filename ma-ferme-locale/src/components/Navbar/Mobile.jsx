import { AppBar, IconButton, Toolbar, Grid, Link } from "@mui/material";
import HomeOutlinedIcon from '@mui/icons-material/HomeOutlined';
import SearchOutlinedIcon from '@mui/icons-material/SearchOutlined';
import PersonOutlineIcon from '@mui/icons-material/PersonOutline';
import React from "react";
import { LogoutOutlined, QuestionMarkOutlined } from "@mui/icons-material";

import user from "../User/User";

export default function Mobile() {

    /* A React hook that is used to set the state of the component. */
    const [clickHex, setClickHex] = React.useState(false);

    return (
        <AppBar position="fixed" color="white" sx={{ top: 'auto', bottom: 0 }}>
            <Toolbar>
                <Grid container justifyContent='space-between'>
                    <Grid item alignSelf='center' >
                        <Link href="/" underline="none" color="inherit">
                            <IconButton color="inherit" aria-label="open drawer">
                                <HomeOutlinedIcon />
                            </IconButton>
                        </Link>
                    </Grid>
                    <Grid item alignSelf='center'>
                        <Link href="/farms" underline="none" color="inherit">
                            <IconButton color="inherit" aria-label="open drawer">
                                <SearchOutlinedIcon />
                            </IconButton>
                        </Link>
                    </Grid>
                    {/*<Grid item sx={{ filter: 'drop-shadow(0px 5px 4px #ccc)' }}>
                        <div class="hexagon" onClick={() => setClickHex(!clickHex)} className={clickHex ? 'hexagon turning' : 'hexagon'}>
                            <AddIcon id="search-icon" color="white" sx={{ display: 'block', margin: 'auto', transform: 'translateY(80%)' }} />
                        </div>
                    </Grid>*/}
                    <Grid item alignSelf='center'>
                        <Link href="/dashboard" underline="none" color="inherit">
                            <IconButton color="inherit" aria-label="open drawer">
                                <PersonOutlineIcon />
                            </IconButton>
                        </Link>
                    </Grid>
                    <Grid item alignSelf='center'>
                        <Link href="/help" underline="none" color="inherit">
                            <IconButton color="inherit" aria-label="open drawer">
                                <QuestionMarkOutlined />
                            </IconButton>
                        </Link>
                    </Grid>
                    <Grid item alignSelf='center'>
                        <IconButton color="inherit" aria-label="open drawer" onClick={() => user.logout()}>
                            <LogoutOutlined />
                        </IconButton>
                    </Grid>
                </Grid>


            </Toolbar>
        </AppBar>
    );
}