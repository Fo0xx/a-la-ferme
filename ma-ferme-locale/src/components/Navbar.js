import React from 'react';

import { AppBar, Box, Toolbar, Typography, IconButton, Menu, Container, Avatar, Button, Tooltip, MenuItem, Link } from '@mui/material';

import '../styles/global.scss';
import user from '../components/User/User';
import { Breakpoint } from 'react-socks';
import Mobile from './Navbar/Mobile';
import { ApiClient } from '../services/ApiClient';
import User from '../components/User/User';
import { Navigate } from 'react-router-dom';

const settings = ['S\'inscrire', 'Se connecter'];

class Navbar extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            anchorElUser: null,
            anchorElNav: null
        }

    }

    handleOpenNavMenu = (event) => {
        this.setState({ anchorElNav: event.currentTarget });
    };
    handleOpenUserMenu = (event) => {
        this.setState({ anchorElUser: event.currentTarget });
    };

    handleCloseNavMenu = () => {
        this.setState({ anchorElNav: null });
    };

    handleCloseUserMenu = () => {
        this.setState({ anchorElUser: null });
    };

    logout = (e) => {
        e.preventDefault();

        User.signOut();

        Navigate('/', { replace: true });

        this.handleCloseUserMenu();
    }

    render() {

        return (
            <>
                <Breakpoint sm up style={{ position: "absolute" }}>
                    <AppBar position="fixed" sx={{ color: 'white', boxShadow: 'none', background: 'linear-gradient(180deg, rgba(0, 0, 0, 0.27) 0%, rgba(255, 255, 255, 0) 100%)' }} >
                        <Container>
                            <Toolbar disableGutters>
                                <Link href="/" underline="none" color="inherit">
                                    <Typography
                                        variant="h6"
                                        noWrap
                                        component="div"
                                        sx={{ mr: 2, display: 'flex' }}
                                    >
                                        MA FERME LOCALE
                                    </Typography>
                                </Link>

                                <Box sx={{ flexGrow: 1, display: 'flex', justifyContent: 'flex-end', mr: 20 }}>
                                    <Button href='/' onClick={this.handleCloseNavMenu} sx={{ my: 2, color: 'inherit', display: 'block' }}>
                                        Accueil
                                    </Button>
                                    <Button href='/farms' onClick={this.handleCloseNavMenu} sx={{ my: 2, color: 'inherit', display: 'block' }}>
                                        Les fermes
                                    </Button>
                                    <Button href='/le-projet' onClick={this.handleCloseNavMenu} sx={{ my: 2, color: 'inherit', display: 'block' }}>
                                        Le projet
                                    </Button>
                                </Box>

                                <Box sx={{ flexGrow: 0 }}>
                                    <Tooltip title="Open settings">
                                        <IconButton onClick={this.handleOpenUserMenu} sx={{ p: 0 }}>
                                            {
                                                user.isLoggedIn() ? <Avatar alt="user" src="/static/images/avatar/2.jpg" /> : <Avatar alt="lambda" src="/static/images/avatar/2.jpg" />
                                            }
                                        </IconButton>
                                    </Tooltip>
                                    <Menu
                                        id="menu-appbar"
                                        anchorEl={this.state.anchorElUser}
                                        anchorOrigin={{
                                            vertical: 'bottom',
                                            horizontal: 'center',
                                        }}
                                        keepMounted
                                        transformOrigin={{
                                            vertical: 'top',
                                            horizontal: 'right'
                                        }}
                                        open={Boolean(this.state.anchorElUser)}
                                        onClose={this.handleCloseUserMenu}
                                    >
                                        {User.isLoggedIn() ?
                                            <div>
                                                <Link href="/dashboard" underline="none" color="inherit">
                                                    <MenuItem
                                                        onClick={this.handleCloseUserMenu}>Mon profil</MenuItem>
                                                </Link>
                                                <MenuItem onClick={this.logout}>Logout</MenuItem>
                                            </div>
                                            :
                                            <>
                                                <Link href="/login" underline="none" color="inherit">
                                                    <MenuItem
                                                        onClick={this.handleCloseUserMenu}>Se connecter</MenuItem>
                                                </Link>
                                                <Link href="/register" underline="none" color="inherit">
                                                    <MenuItem onClick={this.handleCloseUserMenu}>S'inscrire</MenuItem>
                                                </Link>
                                            </>
                                        }

                                    </Menu>
                                </Box>
                            </Toolbar>

                        </Container>
                    </AppBar>
                </Breakpoint>
                <Breakpoint xs only>
                    <Mobile />
                </Breakpoint>
            </>
        );
    }
}

export default Navbar;