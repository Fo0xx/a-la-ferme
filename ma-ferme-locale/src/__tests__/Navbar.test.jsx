import { cleanup, screen, render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import Navbar from '../components/Navbar';

import Enzyme from 'enzyme';
import { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Avatar, IconButton } from '@mui/material';
import Mobile from '../components/Navbar/Mobile';

Enzyme.configure({ adapter: new Adapter() });

describe('Navbar component', () => {

    it('The navbar should render', () => {
        expect(mount(<Navbar />)).toMatchSnapshot();
    });

    //the navbar should render good in mobile
    it('The navbar should render good in mobile', () => {
        expect(mount(<Mobile />)).toMatchSnapshot();
    });

    describe('Navbar render elements correctly', () => {

        let wrapper;

        beforeEach(() => {
            wrapper = shallow(<Navbar />);
        });

        afterEach(cleanup);

        it('The navbar should render the name', () => {
            expect(wrapper.findWhere(el => el.text() === 'MA FERME LOCALE')).toBeTruthy();
        });

        //the navbar should render an Avatar component with the alt 'lambda'
        it('The navbar should render the avatar', () => {
            expect(wrapper.containsMatchingElement(<Avatar />)).toBeTruthy();
        });

        //should render the good avatar if the user is not logged
        it('The navbar should render the good avatar if the user is not logged', () => {
            expect(wrapper.containsMatchingElement(<Avatar alt="lambda" />)).toBeTruthy();
        });

    });

    describe('Navbar render elements correctly on mobile', () => {

        let wrapper;

        beforeEach(() => {
            wrapper = shallow(<Mobile />);
        });

        afterEach(cleanup);

        //the navbar should have 4 icon buttons children
        it('The navbar should have 4 icon buttons children', () => {
            expect(wrapper.find(<IconButton />).length).toBe(5);
        });

    });

    afterEach(cleanup);

});

