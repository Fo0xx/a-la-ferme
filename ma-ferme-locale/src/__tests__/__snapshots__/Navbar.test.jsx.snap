// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component The navbar should render 1`] = `ReactWrapper {}`;

exports[`Navbar component The navbar should render good in mobile 1`] = `ReactWrapper {}`;
